backend/src/routes/__tests__/card.spec.js

```diff
@@ -0,0 +1,529 @@
+        expect(response.body.cost).toBe(card.cost);
+        expect(response.body.attack).toBe(card.attack);
+        expect(response.body.health).toBe(card.health);
+        expect(typeof new Date(response.body.createdAt).toISOString()).toBe('string');
+        expect(response.body.createdAt).toBeDefined();
+        expect(typeof new Date(response.body.updatedAt).toISOString()).toBe('string');
+        expect(response.body.updatedAt).toBeDefined();
+        done();
+      })
+      .catch(done);
+  });
+});
+
+describe('Getting Cards as Unlogged', () => {
+  it('GET /cards/ should return 401', (done) => {
+    request(app)
+      .get('/cards/')
+      .expect(401)
+      .then((response) => {
+        expect(response.body.reason).toBe(undefined);
+        done();
+      })
+      .catch(done);
+  });
+});
+
+describe('Updating a Card PATCH', () => {
+  it('PATCH /cards/:id as an Admin should update the card', (done) => {
+    request(app)
+      .patch(`/cards/${cardId}`)
+      .set('Authorization', `Bearer ${adminToken}`)
+      .send({...updatedCard, name: 'New test'})
+      .expect(200)
+      .expect('Content-Type', /json/)
+      .then((response) => {
+        expect(typeof response.body.id).toBe('number');
+        expect(response.body.name).toBe('New test');
+        expect(response.body.description).toBe(updatedCard.description);
+        expect(response.body.image).toBe(updatedCard.image);
+        expect(response.body.type).toBe(updatedCard.type);
+        expect(response.body.rarity).toBe(updatedCard.rarity);
+        expect(response.body.cost).toBe(updatedCard.cost);
+        expect(response.body.attack).toBe(updatedCard.attack);
+        expect(response.body.health).toBe(updatedCard.health);
+        expect(typeof new Date(response.body.createdAt).toISOString()).toBe('string');
+        expect(response.body.createdAt).toBeDefined();
+        expect(typeof new Date(response.body.updatedAt).toISOString()).toBe('string');
+        expect(response.body.updatedAt).toBeDefined();
+        done();
+      })
+      .catch(done);
+  });
+
+  it('PATCH /cards/:id as a Player should return 403', (done) => {
+    const updatedCard = {
+      name: 'New name',
+      description: 'New description',
+      image: 'https://via.placeholder.com/150'
+    };
+    request(app)
+      .patch(`/cards/${cardId}`)
+      .set('Authorization', `Bearer ${playerToken}`)
+      .send(updatedCard)
+      .expect(403)
+      .then((response) => {
+        expect(response.body.reason).toBe(undefined);
+        done();
+      })
+      .catch(done);
+  });
+  
+  it('PATCH /cards/:id as Unlogged should return 401', (done) => {
+    const updatedCard = {
+      name: 'New name',
+      description: 'New description',
+      image: 'https://via.placeholder.com/150'
+    };
+    request(app)
+      .patch(`/cards/${cardId}`)
+      .send(updatedCard)
+      .expect(401)
+      .then((response) => {
+        expect(response.body.reason).toBe(undefined);
```

Comment:
Il faut plutot vérifier que la réponse entière n'est pas un object.

expect(typeof response.body).not.toBe('object');